---
int8 id
int8 stackNum
bool success